src/components/Global/TokenSelector/TokenSelector.tsx

```
@@ -212,43 +216,42 @@ const TokenSelector = ({
         }))
     }, [selectedChainID, supportedSquidChainsAndTokens, showOnlySquidSupported, shouldBeConnected])
 
-    const _balancesToDisplay = useMemo(() => {
-        let balancesToDisplay: IUserBalance[]
-        const balances = selectedWallet?.balances ?? []
+    useEffect(() => {
+        const fetchBalances = async () => {
+            let balancesToDisplay: IUserBalance[] = []
+            if (!!wagmiAddress) {
```

Biome (1.9.4)

[error] Avoid redundant double-negation.

It is not necessary to use double-negation when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant double-negation


(lint/complexity/noExtraBooleanCast)